build.sbt

```diff
@@ -6,13 +6,14 @@ scalaVersion := "2.11.7"
 
 libraryDependencies += "org.spire-math" %% "cats" % "0.2.0"
 libraryDependencies += "com.lihaoyi" %% "pprint" % "0.3.6"
+libraryDependencies += "com.lihaoyi" %% "upickle" % "0.3.6"
 libraryDependencies += "com.chuusai" %% "shapeless" % "2.2.5"
 
 resolvers += Resolver.sonatypeRepo("releases")
 
 addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.6.3")
 
-wartremoverErrors ++= Warts.allBut(Wart.Any, Wart.Nothing, Wart.Throw, Wart.IsInstanceOf, Wart.AsInstanceOf, Wart.NoNeedForMonad)
+// wartremoverErrors ++= Warts.allBut(Wart.Any, Wart.Nothing, Wart.Throw, Wart.IsInstanceOf, Wart.AsInstanceOf, Wart.NoNeedForMonad)
```

Comment:
TODO: this is removed as there are lots of errors when using upickle. 

a. investigate how to ignore specifically upickle errors
b. swap upickle with an alternative lib
c. disable wart remover :(